api-operator/pkg/apis/wso2/v1alpha1/security_types.go

```diff
@@ -66,14 +66,14 @@ type SecurityList struct {
 
 type SecurityConfig struct {
 	Certificate          string `json:"certificate"`
-	Alias                string `json:"alias"`
-	Endpoint             string `json:"endpoint"`
-	Credentials          string `json:"credentials"`
+	Alias                string `json:"alias,omitempty"`
+	Endpoint             string `json:"endpoint,omitempty"`
+	Credentials          string `json:"credentials,omitempty"`
 	Issuer               string `json:"issuer"`
-	Audience             string `json:"audience"`
+	Audience             string `json:"audience,omitempty"`
 	ValidateSubscription bool   `json:"validateSubscription,omitempty"`
 	ValidateAllowedAPIs  bool   `json:"validateAllowedAPIs,omitempty"`
-	JwksURL              string `json:"jwksURL"`
+	JwksURL              string `json:"jwksURL,omitempty"`
```

Comment:
Shall we make fields `Certificate` and `Issuer` are also optional

- In security type "basic" only required "credentials"
```yaml
apiVersion: wso2.com/v1alpha1
kind: Security
metadata:
  name: petstorebasic
spec:
  type: basic
  securityConfig:
    - credentials: secret-basic
```

Other types
- Oauth
```yaml
apiVersion: wso2.com/v1alpha1
kind: Security
metadata:
  name: petstoreoauth
spec:
  type: Oauth
  securityConfig:
    - certificate: wso2am320-secret #create secret with certificate and add secret name
      endpoint: "https://wso2apim.wso2:9443"
      credentials: oauth-credentials
```

- JWT
```yaml
apiVersion: wso2.com/v1alpha1
kind: Security
metadata:
  name: petstorejwt
spec:
  type: JWT
  securityConfig:
    - issuer: https://wso2apim:32001/oauth2/token
      #create secret with certificate and add secret name
      certificate: wso2am320-secret
      validateSubscription: false
```